### PR TITLE
fix: backgroundThrottling rwh assignment

### PR DESCRIPTION
**Pull request by codebytere**
#### Description of Change

Closes https://github.com/electron/electron/issues/20974.
Follow-up to https://github.com/electron/electron/pull/20239

Previously when a render view was created the corresponding `RenderWidgetHost` was fetched with the wrong routing ID which meant that the `backgroundThrottling` option was not properly set for the RWH and thus the API had no effect.

This PR fixes that.

cc @miniak 

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: Fixed throttling in `webContents.setBackgroundThrottling`.

### CODE REVIEW

File: shell/browser/api/atom_api_web_contents.cc [hunk 1]
```diff
@@ -1286,31 +1286,26 @@ void WebContents::SetBackgroundThrottling(bool allowed) {
...
   auto* render_process_host = render_view_host->GetProcess();
-  if (!render_process_host) {
+  if (!render_process_host)
     return;
-  }
 
-  auto* render_widget_host_impl = content::RenderWidgetHostImpl::FromID(
-      render_process_host->GetID(), render_view_host->GetRoutingID());
-  if (!render_widget_host_impl) {
+  auto* impl = static_cast<content::RenderWidgetHostImpl*>(
```
Comment (deepak1556) [inline shell/browser/api/atom_api_web_contents.cc]:
```suggestion
  auto* render_widget_host_impl = static_cast<content::RenderWidgetHostImpl*>(
```

**Comment by codebytere** [comment]
hmm - looks like this is causing a CHECK failure now 🤔

**Comment by deepak1556** [comment]
Based on https://cs.chromium.org/chromium/src/content/browser/renderer_host/render_widget_host_impl.h?l=818-822 it would make sense that our version would crash given it doesn't check for a valid process when extracting the widget associated with the view.

I think the solution here would be fix the underlying routing id issue by extracting the widget from `RenderFrameHost`. /cc @zcbenz for second opinion.

**Comment by miniak** [comment]
This PR still does not fix #20974. It only fixes `requestAnimationFrame` throttling.

File: shell/browser/api/atom_api_web_contents.cc [hunk 2]
```diff
@@ -1286,31 +1286,26 @@ void WebContents::SetBackgroundThrottling(bool allowed) {
...
     return;
-  }
 
   auto* render_view_host = contents->GetRenderViewHost();
-  if (!render_view_host) {
+  if (!render_view_host)
     return;
-  }
 
   auto* render_process_host = render_view_host->GetProcess();
```
Comment (miniak) [inline shell/browser/api/atom_api_web_contents.cc]: `render_process_host` is not used anymore, please remove

File: shell/browser/api/atom_api_web_contents.cc [hunk 3]
```diff
@@ -1286,31 +1286,26 @@ void WebContents::SetBackgroundThrottling(bool allowed) {
   background_throttling_ = allowed;
 
   auto* contents = web_contents();
```
Comment (miniak) [inline shell/browser/api/atom_api_web_contents.cc]: this is also not necessary, we don't check the return value of `web_contents()` anywhere else

**Comment by deepak1556** [comment]
@miniak I think this PR needs to address https://github.com/electron/electron/pull/21037#issuecomment-551223741 before making any further changes.

**Comment by miniak** [comment]
@deepak1556, @codebytere this issue is quite high priority for us

**Comment by deepak1556** [comment]
@miniak I am working on this today.

**Comment by deepak1556** [comment]
Update: The reason timers aren't affected by this change is because they are controlled separately on the blink end using `blink::PageScheduler` and they throttle based on the page visibility, you can disable this throttling by using `--disable-features="WebContentsOcclusion"` but this will also affect the value of `document.visibilityState`.

I will update the disable_hidden.patch to account for this code path as well without impacting the value of `document.visibilityState`

File: patches/chromium/feat_allow_disbaling_blink_scheduler_throttling_per_renderview.patch [hunk 4]
```diff
@@ -0,0 +1,155 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: deepak1556 <hop2deep@gmail.com>
+Date: Fri, 29 Nov 2019 16:08:14 -0800
+Subject: feat: allow disbaling blink scheduler throttling per RenderView
```
Comment (miniak) [inline patches/chromium/feat_allow_disbaling_blink_scheduler_throttling_per_renderview.patch]: typo `disbaling` -> `disabling`

**Comment by deepak1556** [comment]
Please merge this after merging https://github.com/electron/electron/pull/21343 for a clean backport on all tags listed here.

**Comment by deepak1556** [comment]
failing tests are not related, merging.

**Pull request merged**